test/integration_tests/prototype/test_models.py

```diff
@@ -135,3 +138,127 @@ def test_t5_wrapper_checkpoint(self, name) -> None:
 
         output_text = model(test_text, beam_size, max_seq_len)
         self.assertEqual(output_text, expected_text)
+
+
+class TestLoadFromHFCheckpoints(TorchtextTestCase):
+    def check_outputs_of_models(self, our_output, hf_output, config, encoder_only) -> None:
+        if encoder_only:
+            # check that encoder layers match
+            for i in range(config.num_encoder_layers + 1):
+                if i < config.num_encoder_layers:
+                    # self-attention scores
+                    assert torch.equal(
+                        our_output["encoder_sa_scores"][i], hf_output.attentions[i]
+                    ), f"Mismatched self-attention scores for encoder layer {i}"
+                # encoder hidden states
+                assert torch.equal(
+                    our_output["encoder_hidden_states"][i], hf_output.hidden_states[i]
+                ), f"Mismatched hidden states for encoder layer {i}"
+
+        else:
+            # check that encoder layers match
+            for i in range(config.num_encoder_layers + 1):
+                if i < config.num_encoder_layers:
+                    # self-attention scores
+                    assert torch.equal(
+                        our_output["encoder_sa_scores"][i], hf_output.encoder_attentions[i]
+                    ), f"Mismatched self-attention scores for encoder layer {i}"
+                # encoder hidden states
+                assert torch.equal(
+                    our_output["encoder_hidden_states"][i], hf_output.encoder_hidden_states[i]
+                ), f"Mismatched hidden states for encoder layer {i}"
```

Comment:
This logic seems like an exact copy of that above. Can't we just do the check for the decoder layers if it's not an encoder only model?